torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -739,22 +744,13 @@ void InplaceConverter::replaceAttrWithInplaceOps(
     TORCH_INTERNAL_ASSERT(
         n->kind() == prim::GetAttr || n->kind() == prim::SetAttr);
     if (n->kind() == prim::SetAttr) {
-      // Convert SetAttr to inplace op.
-      // Directly convert to index_put_ instead of copy_, since we know expand
-      // is not required for value.
+      // Convert SetAttr to inplace op aten::set_.
       WithInsertPoint guard(n);
-      auto false_val_ = graph_->insertConstant(false);
-      auto dummy_list =
-          graph_->insertNode(graph_->createList(OptionalType::ofTensor(), {}))
-              ->output();
-
-      auto* index_put_node = graph_->create(aten::index_put_, 1);
-      index_put_node->addInput(find_init_val->second);
-      index_put_node->addInput(dummy_list);
-      index_put_node->addInput(n->input(1));
-      index_put_node->addInput(false_val_);
-      index_put_node->setSourceRange(n->sourceRange());
-      index_put_node->insertBefore(n);
+      auto* set_node = graph_->create(aten::set_, 1);
+      set_node->addInput(find_init_val->second);
+      set_node->addInput(n->input(1));
+      set_node->copyMetadata(n);
+      set_node->insertBefore(n);
     } else if (n->kind() == prim::GetAttr) {
```

Comment:
Do we need say `if (==GetAttr)`? the ASSERT above already says it can only be one of the two.

Reply:
hmm.. the assertion guarantees no need for another else branch after this. I feel it's clearer to developers spelling this out here explicitly with a bit of redundancy.

Reply:
Thanks, agree with this.